agent/consul/acl_server.go

```diff
@@ -145,7 +151,7 @@ func (s *Server) ResolveIdentityFromToken(token string) (bool, structs.ACLIdenti
 	index, aclToken, err := s.fsm.State().ACLTokenGetBySecret(nil, token)
 	if err != nil {
 		return true, nil, err
-	} else if aclToken != nil {
+	} else if aclToken != nil && !aclToken.IsExpired(time.Now()) {
```

Comment:
Originally I wanted to put the expiration check inside of the state store to cut down on the number of places that I had to touch. It felt dirty to give the FSM a clock reference (from a "let's not corrupt the FSM by accident" gut feeling) and it felt slightly less dirty to change all of the token FSM functions to take an `asOf time.Time` param (but still felt like it was loading a future foot gun).

But then there's the problem of the identity acl cache. From what I could tell the acl resolver does exclusively return information from the cache in certain circumstances so it seemed to make sense to check the expiration time post-cache-read to avoid treating a token with an expiration time in the past as still existing past it's shelf life.

Reply:
This is using `time.Now()` not the new `s.currentTime`  I guess in practice that doesn't matter since that currently uses time.Time outside of tests but is it confusing? This is one thing about stubbed time that I don't like especially - it's super easy to add back a `time.Now` somewhere and then be in a bit of a mess when we come to test some new functionality and there are now two different clock sources...

Reply:
In a prior version of this PR I'd confined the `clock` to only exist on the `ACL` endpoint construct to limit the clock sprawl and keep it focused on just checking the endpoint validation/filtering logic without resorting to slower tests due to sleeps. This check is not confined to the `ACL` endpoint and did not initially get a clock threaded through.

Reply:
`clock` is gone now.